src/plugins/knex.js

```diff
@@ -2,22 +2,145 @@
 
 const tx = require('./util/promise')
 
-function createPatch (file) {
-  return {
-    name: 'knex',
-    versions: ['>=0.8.0'],
-    file,
-    patch (Builder, tracer, config) {
-      this.wrap(Builder.prototype, 'then', tx.createWrapThen(tracer, config))
-    },
-    unpatch (Builder) {
-      this.unwrap(Builder.prototype, 'then')
+const spanSymbol = '_sfxSpan'
+
+function createWrapBuilder (tracer, config) {
+  return function wrapQueryBuilder (original) {
+    return function queryBuilderWithTrace () {
+      const scope = tracer.scope()
+      const span = scope.active()
+      const builder = original.apply(this, arguments)
+      builder[spanSymbol] = span
```

Comment:
nit: I believe the approach when modifying elements exposed to the user is to use `Object.defineProperty()` to keep them relatively closed.

Reply:
Thanks. Will do that.

Reply:
Done